integrationTests/chainSimulator/common.go

```diff
@@ -115,21 +121,33 @@ func SendTransaction(
 	txResult, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(tx, maxNumOfBlocksToGenerateWhenExecutingTx)
 	*nonce++
 	require.Nil(t, err)
-	require.NotNil(t, txResult)
-	requireNoSignalErrorEvent(t, txResult)
-	require.Equal(t, transaction.TxStatusSuccess, txResult.Status)
 
 	return txResult
 }
 
-func requireNoSignalErrorEvent(t *testing.T, txResult *transaction.ApiTransactionResult) {
+// RequireSuccessfulTransaction require that the transaction doesn't have signal error event
+func RequireSuccessfulTransaction(t *testing.T, txResult *transaction.ApiTransactionResult) {
+	require.NotNil(t, txResult)
+	require.Equal(t, transaction.TxStatusSuccess, txResult.Status)
 	if txResult.Logs != nil && len(txResult.Logs.Events) > 0 {
 		if txResult.Logs.Events[0].Identifier == signalError {
 			require.Fail(t, string(txResult.Logs.Events[0].Topics[1]))
 		}
 	}
 }
 
+// RequireSignalError require that the transaction has specific signal error
+func RequireSignalError(t *testing.T, txResult *transaction.ApiTransactionResult, error string) {
+	require.NotNil(t, txResult)
+	require.Equal(t, transaction.TxStatusSuccess, txResult.Status)
+	if txResult.Logs != nil && len(txResult.Logs.Events) > 0 {
+		if txResult.Logs.Events[0].Identifier != signalError {
```

Comment:
For both here and above, I don't think it's ok to only check events[0]?
What if you have multiple indexed events, first ones are successful, but later on some are failed?

Reply:
Added a loop through events